test/MStrategy.test.ts

```diff
@@ -1455,6 +1463,533 @@ contract<MStrategy, DeployOptions, CustomContext>("MStrategy", function () {
+                            .commitPermissionGrants(this.params.tokens[i]);
+                    }
+
+                    let validatorFactory = await ethers.getContractFactory(
+                        "MockValidator"
+                    );
+                    let validator = await validatorFactory.deploy(
+                        this.protocolGovernance.address
+                    );
+
+                    await this.protocolGovernance
+                        .connect(this.admin)
+                        .stageValidator(
+                            mockSwapRouter.address,
+                            validator.address
+                        );
+                    await this.protocolGovernance
+                        .connect(this.admin)
+                        .stageValidator(this.usdc.address, validator.address);
+                    await this.protocolGovernance
+                        .connect(this.admin)
+                        .stageValidator(this.weth.address, validator.address);
+
+                    await sleep(this.governanceDelay);
+
+                    await this.protocolGovernance
+                        .connect(this.admin)
+                        .commitValidator(mockSwapRouter.address);
+                    await this.protocolGovernance
+                        .connect(this.admin)
+                        .commitValidator(this.usdc.address);
+                    await this.protocolGovernance
+                        .connect(this.admin)
+                        .commitValidator(this.weth.address);
+
+                    let erc20VaultNft = await this.vaultRegistry.nftForVault(
+                        this.params.erc20Vault
+                    );
+                    let moneyVaultNft = await this.vaultRegistry.nftForVault(
+                        this.params.moneyVault
+                    );
+
+                    await this.vaultRegistry
+                        .connect(this.admin)
+                        .adminApprove(subject.address, erc20VaultNft);
+                    await this.vaultRegistry
+                        .connect(this.admin)
+                        .adminApprove(subject.address, moneyVaultNft);
+
+                    let params: SwapToTargetParamsStruct = {
+                        amountIn: BigNumber.from(10 ** 8),
+                        tokens: this.params.tokens,
+                        tokenInIndex: 0,
+                        priceX96: BigNumber.from(1),
+                        erc20Tvl: [
+                            BigNumber.from(10 ** 3),
+                            BigNumber.from(10 ** 8),
+                        ],
+                        pool: mockUniswapV3Pool.address,
+                        router: mockSwapRouter.address,
+                        erc20Vault: this.params.erc20Vault,
+                        moneyVault: this.params.moneyVault,
+                    };
+
+                    await subject.swapToTarget(params, []);
```

Comment:
Add check for calling router_ with external call with given parameters in line [437 ](https://github.com/mellow-finance/mellow-vaults/blob/main/contracts/strategies/MStrategy.sol#L437)

It can be done with handling this with emit in MockSwapRouter in line [18](https://github.com/mellow-finance/mellow-vaults/blob/00b9537233c897941d506905adc4042930fe463f/contracts/test/MockSwapRouter.sol#L18) and 
checks like [that](https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#emitting-events).